### ISSUE TITLE
StatsRecorder Bug

### ISSUE THREAD

**Issue by uurcm16** [coinfoundry/miningcore#610]
### Description

StatsRecorder does not update stats with no miners connected.

### Steps to Reproduce

1. Start pool
2. Mine for a while and let the pool update its stats
3. Stop mining,
4. Wait for a long time,
5. Wait for the pool to update its stats again 

### Expected behavior

Stats should show 0 hash and 0 connected miners.

### Actual behavior

Gets the latest result and does not update stats because of the results returns 0 values
 
https://github.com/coinfoundry/miningcore/blob/1d79333372d82e9b126a6e69a56861afbd2077a3/src/Miningcore/Mining/StatsRecorder.cs#L141-L142

### Reproduces how often

Always.

### Possible Fix

Adding theese lines to end of the mentioned lines

https://github.com/coinfoundry/miningcore/blob/1d79333372d82e9b126a6e69a56861afbd2077a3/src/Miningcore/Mining/StatsRecorder.cs#L146-L165

```
else
    {
        pool.PoolStats.ConnectedMiners = 0;
        pool.PoolStats.PoolHashrate = 0;
        pool.PoolStats.SharesPerSecond = 0;
        messageBus.NotifyHashrateUpdated(pool.Config.Id, 0);
    }
```

**Comment by peysab** [comment]
I can confirm that your solution is working, I tested on my pool: https://persianmine.com/pool/en/

Thanks

**Comment by calebcall** [comment]
I had done the similar in mine for probably 6 months, definitely works.  Submit a PR and I bet Oliver will accept it.  This comes up often.  Having that remaining hash show up on your pools looks bad for the stats site.  You can have lots of hash reported, no blocks are being found, so it dissuades  miners from using your pool.  It also causes confusion because if you still have one reported, someone else connects, they never see the number of miners go to 2, it stays at 1.

**Comment by oliverw** [comment]
Should be fixed in dev by 6102497d44f612716d2b56516b0aeaa6284ffb78, 3c0285d5cbf5a05695a28b79ef7718114a13dfaa and 9a644194500640c9f108d02d5e2d27785e799cda
